extensions/roam-research/src/parse-template.ts

```diff
@@ -16,29 +21,71 @@ const geneAction = (parentUid: string | number, str: string, uid: number) => {
   };
 };
 
+const roamCreateBlockWithDnpAsParent = (dnpUid: string, str: string, uid: number): roamApiSdk.RoamCreateBlock => {
+  return {
+    action: "create-block",
+    location: {
+      // When we use page-title, if page does not exist already, it will create the page
+      "page-title": { "daily-note-page": dnpUid },
```

Comment:
uses new parameter "page-title" to "location" for create-block which ensures that the daily notes page is created if does not exist already